src/analyses/goto_check.cpp

```diff
@@ -1765,6 +1773,15 @@ void goto_checkt::goto_check(
 
       // the LHS might invalidate any assertion
       invalidate(code_assign.lhs());
+
+      if(has_subexpr(i.code.op1(), [](const exprt &expr) {
+           return expr.id() == ID_r_ok || expr.id() == ID_w_ok;
+         }))
+      {
+        auto rw_ok_cond = rw_ok_check(i.code.op1());
+        if(rw_ok_cond.has_value())
+          i.code.op1() = *rw_ok_cond;
```

Comment:
Hang on, so if any subexpr was an r_ok or w_ok then you overwrite the whole RHS? That seems surprising, what's the rationale for blowing away arbitrary AST nodes here?

Reply:
No, `rw_ok_check` will take care of only replace parts of the RHS.